### PR TITLE
Add content setter to useWatchTextFile hook

### PR DESCRIPTION
**Pull request by Conner1115**
Add a `setContent` function to useWatchTextFile so you can mutate the file

### CODE REVIEW

**Review by masad-frost — DISMISSED** [review]
Let's hold off on this for now, people can use `writeFile`, I don't see the value of adding this.

**Comment by Conner1115** [comment]
@lunaroyster ready for your review on this PR

File: src/types/index.ts [hunk 1]
```diff
@@ -46,7 +46,11 @@ export interface WatchFileWatchers {
 }
 
 export interface WatchTextFileWatchers {
-  onReady: (readyArgs: { initialContent: string; version: number }) => void;
+  onReady: (readyArgs: {
+    initialContent: string;
+    version: number;
+    writeChange: (writeChangeArgs: WriteChangeArgs) => Promise<void>;
+  }) => void;
```
Comment (lunaroyster) [inline src/types/index.ts:53]: thanks

**Review by lunaroyster — APPROVED** [review]
thanks, approving this as a net improvement to the hook.

we'll have to refactor this hook anyway when #17 lands, so just be prepared for that

File: src/react/useWatchTextFile.ts [hunk 2]
```diff
@@ -86,6 +103,17 @@ export default function useWatchTextFile({
...
+        ? (args: WriteChangeArgs) => writeChange.current(args)
+        : (_: WriteChangeArgs) => {},
+      replaceContent: watching
+        ? async (text: string) =>
+            await writeChange.current({
+              from: 0,
+              to: text.length,
+              insert: text,
+            })
+        : (_: string) => {},
```
Comment (lunaroyster) [inline src/react/useWatchTextFile.ts:116]: i think this should throw

File: src/react/useWatchTextFile.ts [hunk 3]
```diff
@@ -86,6 +103,17 @@ export default function useWatchTextFile({
       content,
       watching,
       watchError,
+      writeChange: watching
+        ? (args: WriteChangeArgs) => writeChange.current(args)
+        : (_: WriteChangeArgs) => {},
```
Comment (lunaroyster) [inline src/react/useWatchTextFile.ts:108]: same here

**Review by lunaroyster — APPROVED** [review]

**Comment by lunaroyster** [comment]
> Let's hold off on this for now, people can use writeFile, I don't see the value of adding this.

@masad-frost i think it's useful, it would replace most of the code in the JSON editor example right now (even if we're yet to make it fully correct with #17)

**Review by masad-frost — COMMENTED** [review]
We should at least make it work as expected, update the text in state

**Pull request merged**